src/object/pot.c

```diff
@@ -163,7 +167,137 @@ void sub_080825F0(Entity* this) {
     this->action = 1;
     this->previousActionFlag = 0;
 }
+
+void sub_08082614(Entity* this) {
+    u32 tileType;
+
+    sub_0800445C(this);
+    if (--this->actionDelay != 0xFF) {
+        sub_0806F69C(this);
+        sub_08016A6C(this);
+        return;
+    }
+
+    if (sub_0800442E(this)) {
+        return;
+    }
+
+    this->action = 1;
+    if (gPlayerState.flags.all & 0x80) {
+        this->nonPlanarMovement <<= 1;
+    }
+
+    this->field_0x70.HALF.LO = sub_080001DA(COORD_TO_TILE(this), this->collisionLayer);
+    tileType = GetTileTypeByEntity(this);
+    switch (tileType) {
+        case 0x71:
+        case 0x72:
+            sub_08082850(this, NULL);
+            break;
+        default:
+            SetTile(0x4000, COORD_TO_TILE(this), this->collisionLayer);
+            sub_08078930(this);
+            break;
+    }
+}
+
+void sub_080826E4(Entity* this) {
+    gUnk_0811F0C4[this->previousActionFlag](this);
+}
+
+void sub_080826FC(Entity* this) {
+    this->previousActionFlag = 1;
+    this->field_0x1d = 48;
+    this->actionDelay = 0;
+}
+
+void sub_0808270C(Entity* this) {
+    if ((gPlayerState.field_0x1c & 0xF) != 0x1 || (this->bitfield & 0x7F) != 0x13) {
+        this->spriteOffsetX = 0;
+        this->action = 1;
+        SetTile(0x4000, COORD_TO_TILE(this), this->collisionLayer);
+    } else {
+        sub_0806F4E8(this);
+    }
+}
+
+void sub_08082778(Entity* this) {
+    if (this->actionDelay == 0) {
+        this->actionDelay = 1;
+        this->spriteOffsetX = 0;
+        this->spriteOffsetY = -2;
+        SetTile((u16)this->field_0x70.HALF.LO, COORD_TO_TILE(this), this->collisionLayer);
+    }
+
+    if ((gPlayerState.field_0x1c & 0xF) != 0x1 || (this->bitfield & 0x7F) != 0x13) {
+        sub_08082850(this, NULL);
+    } else {
+        sub_0806F3E4(this);
+    }
+}
+
+void sub_080827F8(Entity* this) {
+    if (gPlayerState.field_0x1c == 0) {
+        sub_08082850(this, NULL);
+    }
+}
+
+void nullsub_512(Entity* this) {
+}
+
+void sub_08082818(Entity* this) {
+    sub_08082850(this, NULL);
+}
+
+static void sub_08082824(Entity* this) {
+    if (this->field_0x20 < 0) {
+        this->spriteSettings.b.flipY = 1;
+    }
+
+    if (sub_08003FC4(this, 0x2000) == 0) {
+        sub_08082850(this, NULL);
+    }
+}
+
+static void sub_08082850(Entity* this, Entity* parent) {
+    u32 parameter = sub_0808288C(this, this->entityType.form, this->field_0x7c.BYTES.byte1, this->entityType.parameter);
+    Entity* fxEntity = CreateFx(this, 5, parameter);
+    if (fxEntity) {
+        fxEntity->parent = parent;
+    }
+
+    if (this->entityType.parameter == 1) {
+        SetFlag(this->field_0x86);
+    }
+
+    DeleteThisEntity();
+}
+
+u32 sub_0808288C(Entity* this, u32 form, u32 arg2, u32 arg3) {
+    Entity* entity;
+    u32 result = 0;
+    switch (form) {
+        case 0xFF:
+            result = 0;
+            break;
+        case 0:
+            result = 0x80;
+            break;
+            break;
```

Comment:
BROKEN

Reply:
whoops, fixed

Reply:
@jiangzhengwenjz Since you don't appear to be on Discord anymore, could you explain how the individual `.incbin` statements in the `data/` directory were generated?  Every one I checked didn't have any references in `asm/` or `src/` code.  I'm wondering if it makes sense to collapse them into single large `.incbin` statements for now, unless there was good logic for the way they currently exist.

Reply:
IIRC, I examined every 4-byte aligned word in data region to see if it's possibly a pointer to somewhere also in data region, and if so I create a label for it. 
But clearly it won't work on any pointer that is not aligned to 4-byte boundary (e.g. script, if any) and it can have false positives (e.g. some data that occasionally looks like a pointer). 

BTW I'm still on discord, just not in any server.